docs/reference/python/conventions.md

```diff
@@ -18,6 +18,102 @@ pep8 style guide.
     Our CI automatically checks that your code follows the pep8 standard. If it does not, your pull requests will
     be blocked from being merged until those issues are resolved!
 
+### Type hinting
+
+Even though Python is a dynamically typed language, newer versions support type hinting. Type hinting catches
+errors, documents code, improves IDEs and linters, and helps build and maintain a clean software architecture.[^1]
+Expanding on how it catches errors, a static type checker such as [`mypy`](https://mypy.readthedocs.io/en/stable/index.html){target=_blank}
+can be used.
+
+There are [pros and cons](https://realpython.com/lessons/pros-and-cons-type-hints/){target=_blank} to using type
+hinting, make sure that you are aware of them before deciding to utilize them in your codebase.
```

Comment:
Not sure if this sentence is necessary:

- The link is cited at the bottom of the page
- Target audience is firstly sailbot members, who aren't the ones to decide whether to use type hints